test/test.go

```diff
@@ -524,9 +543,18 @@ func (h *Helper) Exist(path string) bool {
 
 // MustNotExist fails if path exists.
 func (h *Helper) MustNotExist(path string) {
+	if err := h.ShouldNotExist(path); err != nil {
+		h.t.Fatalf("%+v", err)
+	}
+}
+
+// ShouldNotExist returns an error if path exists.
+func (h *Helper) ShouldNotExist(path string) error {
```

Comment:
I'm not sure about this pattern - `Should{Not}Exist()` vs. `Must{Not}Exist()`. It seems to mirror the distinction between `testing.T.Error()` and `testing.T.Fatal()`, but it actually doesn't, because here it's left up to the caller to actually set an error. I think that might be misleading.

I can see why you're doing it this way - you want to be able to compose it with higher-order calls. But let's compose these differently - have the function that returns the error be a non-exported one, called by all of these helper methods.